src/state/apis/nft/nftApi.ts

```diff
@@ -180,34 +182,36 @@ export const nftApi = createApi({
     getNftCollection: build.query<NftCollectionType, GetNftCollectionInput>({
       queryFn: async ({ collectionId, accountIds }, { dispatch }) => {
         try {
-          const { assetReference: collectionAddress, chainId } = fromAssetId(collectionId)
-          const alchemyCollectionData = await getAlchemyInstanceByChainId(chainId)
-            .nft.getContractMetadata(collectionAddress)
-            .then(contract => parseAlchemyNftContractToCollectionItem(contract, chainId))
-
-          // Alchemy is the most/only reliable source for collection data for now
-          if (alchemyCollectionData) {
-            dispatch(nft.actions.upsertCollection(alchemyCollectionData))
-            return { data: alchemyCollectionData }
-          }
+          const services = [
+            getAlchemyCollectionData,
+            (collectionId: AssetId, accountIds: AccountId[]) =>
+              dispatch(
+                zapperApi.endpoints.getZapperCollectionBalance.initiate({
+                  accountIds,
+                  collectionId,
+                }),
+              ),
+          ]
 
-          // Note, this will consistently fail, as getZapperCollectionBalance is monkey patched for the RTK query to reject
-          // The reason for that is Zapper is currently rugged upstream - if we don't get Alchemy data, we're out of luck and can't get meta
-          const { data: zapperCollectionData } = await dispatch(
-            zapperApi.endpoints.getZapperCollectionBalance.initiate({ accountIds, collectionId }),
+          const results = await Promise.all(
```

Comment:
should we `allSettled` to get double the failure tolerance, or are both somehow required here?

Reply:
Very valid point, `getAlchemyCollectionData` isn't a RTK query thunk so it *can* reject - a824369bfab95e0c0dc903754d17c9724e00f6ac